modules/eks-managed-node-group/main.tf

```diff
@@ -34,7 +34,7 @@ locals {
 }
 
 resource "aws_launch_template" "this" {
-  count = var.create && var.create_launch_template && var.use_custom_launch_template ? 1 : 0
+  count = var.create && var.create_launch_template && !var.use_custom_launch_template ? 1 : 0
```

Comment:
thats not quite what that variable means - if you want to disable the creation of the launch template, then you set `create_launch_template = false`